flowmachine/flowmachine/features/location/majority_location.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+from flowmachine.features.utilities.subscriber_locations import BaseLocation
+
+
+class MajorityLocation(BaseLocation, Query):
+    def __init__(self, subscriber_subset: Query, weight_column):
+        if weight_column not in subscriber_subset.column_names:
+            raise ValueError("weight_column must exist in subscriber_subset")
+        self.subscriber_subset = subscriber_subset
+        self.weight_column = weight_column
```

Comment:
I think it would be worth doing `self.spatial_unit = subscriber_subset.spatial_unit` here - other queries that accept this one as a sub-query will often look for that attribute.